src/main/java/com/google/sps/PastDeeds/PastDeedsServlet.java

```diff
@@ -0,0 +1,89 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.testing;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.datastore.Key;
+
+import com.google.gson.Gson;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.ArrayList;
+
+import com.google.sps.testing.GoodDeed;
+
+@WebServlet("/ghost-of-deeds-past")
+public class PastDeedsServlet extends HttpServlet {
+    private static final String FALSE = "false";
+    private static final String TRUE = "true";
+    private static final String GOOD_DEED = "GoodDeed";
+    private static final String NAME = "Name";
+    private static final String DESCRIPTION = "Description";
+    private static final String POSTED_YET = "Posted Yet";
+    private static final String LINK = "Link";
+    private static final String TIME_STAMP = "Timestamp";
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        // Fetch Posted Deeds
+        List<GoodDeed> deeds = fetchPostedDeeds();
+
+        Gson gson = new Gson();
```

Comment:
minor - this can be a member variable so that we don't create it each time the method is called